scripts/jenkins/Jenkinsfile

```diff
@@ -415,6 +421,52 @@ kill `pidof dbus-daemon`
     }
 }
 
+/* Generate Stages that build Debian packages
+ */
+def generate_package_build_stages() {
+    def tasks = [:]
+    tasks << build_package_debian_stretch()
+    return tasks
+}
+
+def build_package_debian_stretch() {
+    return withDockerEnv("package/stretch", DOCKER_IMAGES.stretch) {
+        withCredentials([file(credentialsId: 'jenkins-key', variable: 'KEY'),
+                         file(credentialsId: 'jenkins-secret-key', variable: 'SKEY')]) {
+            sh "gpg --import $KEY"
+            sh "gpg --import $SKEY"
+        }
+        withEnv(["DEBSIGN_PROGRAM=gpg",
+                 "DEBFULLNAME=Jenkins (User for Elektra automated build system)",
+                 "DEBEMAIL=autobuilder@libelektra.org"]) {
+            sh "rm -R ./*"
+            def target_dir="./libelektra"
+            checkout scm: [
+                $class: 'GitSCM',
+                branches: scm.branches,
+                extensions: scm.extensions + [
+                    [$class: 'PerBuildTag'],
```

Comment:
What do all these extensions do? Can you point to the docu of where you found all these knobs?

Reply:
googling jenkinsfile reference brings the full list available via https://jenkins.io/doc/pipeline/steps/

Reply:
Any reason why the file is not visible without login?

Yes, such things should be in the intro of the file.

Reply:
> Any reason why the file is not visible without login?

No Idea.

we don't link c tutorials in c files, we don't link manpages in our scripts, what makes the Jenkinsfile so different?

Reply:
> we don't link c tutorials in c files, we don't link manpages in our scripts,

Because it is not one of the development languages and nevertheless nearly everybody will have to do something with it (with minimal knowledge of the language).

And the Jenkinsfile had little docu even compared to C code. But it slowly improves. :wink: 

> what makes the Jenkinsfile so different?

If I understood you correctly, Jenkinsfile might be written in either groovy or in a DSL. So you need to specify which one we use now.

Reply:
it will always be 'groovyish', the choice is between a declarative and a 'scripted' version of the DSL. we currenlty use the scripted one because the declarative has been not complex enough.

However they are to some degree interchangeable.

> Because it is not one of the development languages and nevertheless nearly everybody will have to do something with it (with minimal knowledge of the language).

The problem with documenting essential language features is that it is very much a moving target. Patches come out at least once a week.

Linking the documentation that can be found via simple google queries is not providing any benefit, else you would add c tutorials to every c header, or cmake documentation to every cmakelists file.

I would prefer to stop having those documentation discussions till I have finished the accompanying doc/*.md file.

Reply:
> it will always be 'groovyish', the choice is between a declarative and a 'scripted' version of the DSL. we currenlty use the scripted one because the declarative has been not complex enough.

Seems like I misunderstood. Please add this information.

> Linking the documentation that can be found via simple google queries is not providing any benefit, else you would add c tutorials to every c header, or cmake documentation to every cmakelists file.

No, not every file, only at the entry points. I added them now for cmake 9a7dcbe9b192ebab62b407b6fb010c0773651f4b

> you would add c tutorials

There is a huge difference about how well known C and groovy are. What applies to groovy might not apply to C. CMake is similar, though.

> I would prefer to stop having those documentation discussions till I have finished the accompanying doc/*.md file.

Ok. But please makes notes of these discussion because the next time I review I will most likely overlook things. And newcomers (speak everyone from the current dev team because afaik nobody else knows these DSLs) will be happy for every pointer that is necessary for them.